test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -806,6 +819,364 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
+
+				By("Waiting when mcp finishes updates")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+				annotations := map[string]string{
+					"cpu-c-states.crio.io":      "enable",
+					"cpu-freq-governor.crio.io": "schedutil",
+				}
+
+				cpuCount := "2"
+				resCpu := resource.MustParse(cpuCount)
+				resMem := resource.MustParse("100Mi")
+				testpod = pods.GetTestPod()
+				testpod.Namespace = testutils.NamespaceTesting
+				testpod.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resCpu,
+						corev1.ResourceMemory: resMem,
+					},
+				}
+				testpod.Spec.NodeSelector = map[string]string{testutils.LabelHostname: workerRTNodes[0].Name}
+				testpod.Annotations = annotations
+				runtimeClass := components.GetComponentName(profile.Name, components.ComponentNamePrefix)
+				testpod.Spec.RuntimeClassName = &runtimeClass
+
+				By("creating test pod")
+				err = testclient.Client.Create(context.TODO(), testpod)
+				Expect(err).ToNot(HaveOccurred())
+				err = pods.WaitForCondition(testpod, corev1.PodReady, corev1.ConditionTrue, 10*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testpod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed), "Test pod does not have QoS class of Guaranteed")
+
+				By("Getting the container cpuset.cpus cgroup")
+				containerID, err := pods.GetContainerIDByName(testpod, "test")
+				Expect(err).ToNot(HaveOccurred())
+
+				containerCgroup := ""
+				Eventually(func() string {
+					cmd := []string{"/bin/bash", "-c", fmt.Sprintf("find /rootfs/sys/fs/cgroup/cpuset/ -name *%s*", containerID)}
+					containerCgroup, err = nodes.ExecCommandOnNode(cmd, &workerRTNodes[0])
+					Expect(err).ToNot(HaveOccurred())
+					return containerCgroup
+				}, (cluster.ComputeTestTimeout(30*time.Second, RunningOnSingleNode)), 5*time.Second).ShouldNot(BeEmpty(),
+					fmt.Sprintf("cannot find cgroup for container %q", containerID))
+
+				By("Checking what CPU the pod is using")
+				cmd := []string{"/bin/bash", "-c", fmt.Sprintf("cat %s/cpuset.cpus", containerCgroup)}
+				output, err := nodes.ExecCommandOnNode(cmd, &workerRTNodes[0])
+				Expect(err).ToNot(HaveOccurred())
+				cpus, err := cpuset.Parse(output)
+				for _, cpu := range cpus.ToSlice() {
+					cmd := []string{"/bin/bash", "-c", fmt.Sprintf("cat /sys/devices/system/cpu/cpu%d/power/pm_qos_resume_latency_us", cpu)}
+					output, err := nodes.ExecCommandOnNode(cmd, &workerRTNodes[0])
+					Expect(err).ToNot(HaveOccurred())
+					Expect(output).To(Equal("0"))
+					cmd = []string{"/bin/bash", "-c", fmt.Sprintf("cat /sys/devices/system/cpu/cpu%d/power/cpufreq/scaling_governor", cpu)}
+					output, err = nodes.ExecCommandOnNode(cmd, &workerRTNodes[0])
+					Expect(output).To(Equal("schedutil"))
+					Expect(err).ToNot(HaveOccurred())
+				}
+				deleteTestPod(testpod)
```

Comment:
After the pod is deleted, it would be good to check that the cpus that were used by the pod have been set back to their original pm_qos_resume_latency and scaling_governor.

Reply:
ack, will add that

Reply:
@bartwensley i have addressed this in the latest patch